client/src/main/kotlin/io/spine/examples/shareaware/client/MarketPage.kt

```diff
@@ -297,63 +338,486 @@ public class MarketPageModel(client: DesktopClient) {
 @OptIn(ExperimentalMaterial3Api::class)
 @Composable
 public fun MarketPage(model: MarketPageModel) {
+    val marketShares by model.shares().collectAsState()
+    val previousShares by model.previousShares().collectAsState()
+    val selectedShareId by model.selectedShare().collectAsState()
     val popUpShown = model.purchaseOperation.isResultMessageShown().collectAsState()
     val popUpMessage = model.purchaseOperation.resultMessage().collectAsState()
     val popUpInErrorState = model.purchaseOperation.isFailed().collectAsState()
     val popUpContentColor = if (popUpInErrorState.value) MaterialTheme.colorScheme.error
     else MaterialTheme.colorScheme.primary
-    Scaffold(
-        bottomBar = {
-            PopUpMessage(
-                isShown = popUpShown.value,
-                dismissAction = { model.purchaseOperation.closeOperationResultMessage() },
-                label = popUpMessage.value,
-                contentColor = popUpContentColor,
-                modifier = Modifier
-                    .wrapContentWidth()
-                    .zIndex(1f)
-            )
+    val purchaseInProgress = model.purchaseOperation.isInProgress().collectAsState()
+    val listState = rememberLazyListState()
+    Row(
```

Comment:
When you see the content which takes more than one page, this is a sign it needs to be re-formatted into a number of components or functions.

Similar to my comment above, please rework into smaller meaningful portions.